src/Fields/field_tuples.jl

```diff
@@ -55,9 +108,9 @@ can be specified to assign data arrays to each tendency field.
 """
 function TendencyFields(arch, grid, tracer_names; kwargs...)
     velocities = (
-        u = :u ∈ keys(kwargs) ? XFaceField(arch, grid, kwargs[:u]) : XFaceField(arch, grid),
-        v = :v ∈ keys(kwargs) ? YFaceField(arch, grid, kwargs[:v]) : YFaceField(arch, grid),
-        w = :w ∈ keys(kwargs) ? ZFaceField(arch, grid, kwargs[:w]) : ZFaceField(arch, grid)
+        u = :u ∈ keys(kwargs) ? kwargs[:u] : XFaceField(arch, grid, UVelocityBoundaryConditions(grid), zeros(arch, grid)),
```

Comment:
These can be defined in the constructor, avoiding the chain of if statements:

```julia
function TendencyFields(arch, grid, tracer_names; 
    u = XFaceField(arch, grid, UVelocityBoundaryConditions(grid), zeros(arch, grid),
    # similar for v and w
    tracer_kwargs...)
```

Reply:
Maybe I misinterpreted what is happening here. Either way the 3 if statements indicate that a better design is possible and should be pursued.

Reply:
I've now defined u, v, w in the function signature and got rid of the ternary conditionals.